### PR TITLE
Write that follows timed out write always hangs

### PR DESCRIPTION
**Pull request by dshulyak**
Write on the same fd that follows write that was canceled with IORING_OP_LINK_TIMEOUT never completes. 
See attached tests for repro.

Was discovered on 5.7.15

### CODE REVIEW

**Comment by isilence** [comment]
This one was pretty useful. Are you going to reopen it later?
I'll comment on a coupe of thing there anyway.

File: test/link-timeout.c [hunk 1]
```diff
@@ -992,6 +992,106 @@ static int test_timeout_link_chain5(struct io_uring *ring)
...
+			if (cqe->res != -ETIME) {
+				fprintf(stderr, "got %d, wanted -ETIME\n", cqe->res);
+				goto err;
+			}
+			break;
+		}
+		io_uring_cqe_seen(ring, cqe);
+	}
+
+	fprintf(stderr, "timeout + write were completed\n");
```
Comment (isilence) [inline test/link-timeout.c:1064]: Please, remove non-error printf()'s

File: test/link-timeout.c [hunk 2]
```diff
@@ -992,6 +992,106 @@ static int test_timeout_link_chain5(struct io_uring *ring)
...
+
+		switch (cqe->user_data) {
+		case 1:
+			if (cqe->res != -ECANCELED) {
+				fprintf(stderr, "write got %d, wanted -ECANCELED\n", cqe->res);
+				goto err;
+			}
+			break;
+		case 2:
+			if (cqe->res != -ETIME) {
```
Comment (isilence) [inline test/link-timeout.c:1055]: The write may actually complete before the timeout had a chance
to cancel it, so it fails for me from time to time.

Comment (dshulyak) [reply test/link-timeout.c:1055]: i noticed this as well. i have tried to use timespec with s=0, ns=0, but it doesn't seem to work as i expected

Comment (axboe) [reply test/link-timeout.c:1055]: -EALREADY is probably also a valid return here.

**Comment by axboe** [comment]
> This one was pretty useful. Are you going to reopen it later?

Agree, I think we should add the test case, just needs a bit of cleaning to adhere to what is expected of the regression tests.

**Pull request closed without merging**